### PR TITLE
MAILBOX-381 API redesign: Separate content from meta-data

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java [hunk 1]
```diff
@@ -19,12 +19,16 @@
...
 
+import java.io.InputStream;
+
 import org.apache.james.core.User;
 import org.apache.james.mailbox.model.MessageId;
 import org.reactivestreams.Publisher;
 
 public interface DeletedMessageVault {
-    Publisher<Void> append(User user, DeletedMessage deletedMessage);
+    Publisher<Void> append(User user, DeletedMessage deletedMessage, InputStream data);
```
Comment (mbaechler) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java]: s/data/mimeMessage/ ?

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java [hunk 2]
```diff
@@ -19,12 +19,16 @@
...
+
+    Publisher<InputStream> loadContent(User user, MessageId messageId);
```
Comment (mbaechler) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java]: s/loadContent/loadMimeMessage?

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java [hunk 3]
```diff
@@ -34,18 +40,25 @@
 
 public class MemoryDeletedMessagesVault implements DeletedMessageVault {
     private final Table<User, MessageId, DeletedMessage> table;
+    private final Table<User, MessageId, byte[]> contentTable;
```
Comment (mbaechler) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java]: I'd rather put a Tuple in the previous table. It makes it simpler to handle atomicity

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java [hunk 4]
```diff
@@ -34,18 +40,25 @@
...
 
     MemoryDeletedMessagesVault() {
         table = HashBasedTable.create();
+        contentTable = HashBasedTable.create();
     }
 
     @Override
-    public synchronized Mono<Void> append(User user, DeletedMessage deletedMessage) {
+    public synchronized Mono<Void> append(User user, DeletedMessage deletedMessage, InputStream inputStream) {
```
Comment (mbaechler) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java]: it's usually better to lock (synchronize) on a private object. With the current code, anybody putting a lock on this object will screw your locking logic.

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java (hunk 1 above)
Comment (mbaechler) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java]: not fixed

**Comment by Arsnael** [comment]
Read it...

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageVault.java]: Sorry

File: mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultSearchContract.java [hunk 5]
```diff
@@ -588,7 +587,7 @@ default DeletedMessage storeMessageWithSender(MaybeSender sender) {
             .deletionDate(DELETION_DATE)
             .sender(sender)
             .recipients(RECIPIENT1, RECIPIENT2)
-            .content(() -> new ByteArrayInputStream(CONTENT))
+
```
Comment (blackheaven) [inline mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultSearchContract.java]: Why an extra line here?

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**